Allow WORD_TERM regex match to execute before Func

ammarv23 opened the pull request:
Running into an issue with an escaped string causing a catastrophic
backtrack with the regex to determine the CSS_TOKEN. The offending string
is matched by the WORD_TERM portion of the regex though, so by moving it ahead
of FUNCTION in the join, we can ensure we perform the quicker match before
hitting the catastrophy

Commit:
Allow WORD_TERM regex match to execute before Func

Running into an issue with an escaped string causing a catastrophic
backtrack with the regex to determine the CSS_TOKEN. The offending string
is matched by the WORD_TERM portion of the regex though, so by moving it ahead
of FUNCTION in the join, we can ensure we perform the quicker match before
hitting the catastrophy

jasonmclaren reviewed (changes requested):
Good job figuring this out.

Would be nice to see a couple of unit tests:
- can we sanitize the nasty word
- can we still sanitize functions

Commit:
Add test for escaped string

jasonmclaren reviewed (approved):
It's disturbing that the function token can blow up like that... this is frankly the worst regexp I've ever read.

Approved!

Commit:
Improve Function regex

Also revert swap of WORD_TERM and FUNCTION clauses in the regex, turns out
this ordering is important to avoid breaking the rendering of colors defined as functions
(i.e rgb(0, 0, 0)). This new lookahead will prevent catastrophic backtracking by ensuring the
full group is matched before attempting to find the opening brace
